images/router/haproxy/reload-haproxy

```diff
@@ -67,6 +67,26 @@ function haproxyHealthCheck() {
 
 old_pids=$(pidof haproxy)
 
+# If signaled, stop accepting new connections and drain the current processes
+if [ -n "${ROUTER_SHUTDOWN-}" ]; then
+  echo " - Shutting down"
+  if [ -z "$old_pids" ]; then
+    exit 0
+  fi
+  readonly wait_time=${ROUTER_MAX_SHUTDOWN_TIMEOUT:-${MAX_RELOAD_WAIT_TIME:-$max_wait_time}}
+  kill -USR1 $old_pids
+  for i in $( seq 1 $wait_time ); do
```

Comment:
Hrm, that construct looks much harder to understand for a casual reader.  Why is it more precise?

Reply:
Because the condition is true once `$wait_time` seconds has passed since the assignment.  In contrast, sleeping for 1 second _n_ times fails to account for execution time.  It probably does not matter here as the only commands are builtins and `pidof`, which _should_ be fast; it would matter more for a command like, say, `curl`, where the loop may spend more time executing commands than sleeping.  That is why I say it probably does not matter enough to worry about, and if the `for` plus `seq` is easier to read, we can run with that.

Reply:
Yeah, I'd prefer to use the simpler construct if we don't have a concrete reason to avoid it (waiting slightly longer should not be an issue).